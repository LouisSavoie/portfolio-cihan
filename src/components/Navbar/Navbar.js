import React from 'react'
import Container from 'react-bootstrap/Container'
import BSNavbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav'

const Navbar = () => {
  return (
    <BSNavbar data-testid="navbar" collapseOnSelect expand="sm" bg="dark" variant="dark" fixed="top">
      <Container>
      <BSNavbar.Brand>Placeholder</BSNavbar.Brand>
      <BSNavbar.Toggle aria-controls="responsive-navbar-nav" />
      <BSNavbar.Collapse id="responsive-navbar-nav">
        <Nav className="me-auto">
        </Nav>
        <Nav>
          <Nav.Link href="#info">Info</Nav.Link>
          <Nav.Link href="#projects">Projects</Nav.Link>
          <Nav.Link href="#blog">Blog</Nav.Link>
        </Nav>
      </BSNavbar.Collapse>
      </Container>
    </BSNavbar>
  )
}

export default Navbar
