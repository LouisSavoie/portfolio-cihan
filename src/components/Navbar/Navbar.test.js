import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import Navbar from './Navbar'

afterEach(() => {
  cleanup()
})

test('should render Navbar', () => {
  render(<Navbar />)
  const NavbarElement = screen.getByTestId('navbar')
  expect(NavbarElement).toBeInTheDocument()
})

test('matches snapshot', () => {
  const tree = renderer.create(<Navbar />).toJSON()
  expect(tree).toMatchSnapshot()
})
