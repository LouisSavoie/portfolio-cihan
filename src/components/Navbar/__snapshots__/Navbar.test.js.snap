// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark fixed-top"
  data-testid="navbar"
>
  <div
    className="container"
  >
    <span
      className="navbar-brand"
    >
      Placeholder
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="me-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key="#info"
          disabled={false}
          href="#info"
          onClick={[Function]}
        >
          Info
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="#projects"
          disabled={false}
          href="#projects"
          onClick={[Function]}
        >
          Projects
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="#blog"
          disabled={false}
          href="#blog"
          onClick={[Function]}
        >
          Blog
        </a>
      </div>
    </div>
  </div>
</nav>
`;
